### PR TITLE
new module: gatk4/splitintervals

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/gatk4/splitintervals/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = fasta ? "--reference $fasta" : ""
+
+    def avail_mem = 3
+    if (!task.memory) {
+        log.info '[GATK IntervalListToBed] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
```
Comment (matthdsm) [inline modules/gatk4/splitintervals/main.nf:30]: fix program name

File: tests/modules/gatk4/splitintervals/test.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  tags:
+    - gatk4/splitintervals
+    - gatk4
+  files:
+    - path: output/gatk4/test/0000-scattered.interval_list
+      md5sum: c8d6b19e7a92535b6ce9608eae558faa
+    - path: output/gatk4/test/0001-scattered.interval_list
+      md5sum: b1877ad96aec308906594c50ebbe3ded
+    - path: output/gatk4/versions.yml
+      md5sum: c7b9068d84e0e05412f50e86d5b87078
```
Comment (matthdsm) [inline tests/modules/gatk4/splitintervals/test.yml:12]: remove versions checksum

**Review by matthdsm — CHANGES_REQUESTED** [review]

**Review by matthdsm — APPROVED** [review]
LGTM if tests pass

File: modules/gatk4/splitintervals/main.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    input:
+    tuple val(meta), path(intervals)
+    path(fasta)
+    path(fasta_fai)
+    path(dict)
+
+    output:
+    tuple val(meta), path("**.interval_list"), emit: split_intervals
+    path "versions.yml"           , emit: versions
```
Comment (FriederikeHanssen) [inline modules/gatk4/splitintervals/main.nf:18]: align `,`

File: modules/gatk4/splitintervals/main.nf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    } else {
+        avail_mem = task.memory.giga
+    }
+
+    """
+    gatk --java-options "-Xmx${avail_mem}g" SplitIntervals \\
+        --output ${prefix} \\
+        --intervals $intervals \\
+        $reference \\
+        $args
```
Comment (FriederikeHanssen) [inline modules/gatk4/splitintervals/main.nf:40]: add `--tmp-dir . \\` We had weird effects in the past when telling nextflow to write tmp files elsewhere and GTAK tools stil writing to `/tmp`, so decided to set this always.

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**